mm2src/db_common/src/async_sql_conn.rs

```diff
@@ -0,0 +1,288 @@
+    ///
+    /// This method can cause a `panic` if the underlying database connection is closed.
+    /// it is a more user-friendly alternative to the [`AsyncConnection::call`] method.
+    /// It should be safe if the connection is never explicitly closed (using the [`AsyncConnection::close`] call).
+    ///
+    /// Calling this on a closed connection will cause a `panic`.
+    pub async fn call_unwrap<F, R>(&self, function: F) -> R
+    where
+        F: FnOnce(&mut rusqlite::Connection) -> R + Send + 'static,
+        R: Send + 'static,
+    {
+        let (sender, receiver) = oneshot::channel::<R>();
+
+        self.sender
+            .send(Message::Execute(Box::new(move |conn| {
+                let value = function(conn);
+                let _ = sender.send(value);
+            })))
+            .expect("database connection should be open");
+
+        receiver.await.expect("Bug occurred, please report")
+    }
+
+    /// Close the database AsyncConnection.
+    ///
+    /// This is functionally equivalent to the `Drop` implementation for
+    /// `AsyncConnection`. It consumes the `AsyncConnection`, but on error returns it
+    /// to the caller for retry purposes.
+    ///
+    /// If successful, any following `close` operations performed
+    /// on `AsyncConnection` copies will succeed immediately.
+    ///
+    /// On the other hand, any calls to [`AsyncConnection::call`] will return a [`AsyncConnError::ConnectionClosed`],
+    /// and any calls to [`AsyncConnection::call_unwrap`] will cause a `panic`.
+    ///
+    /// # Failure
+    ///
+    /// Will return `Err` if the underlying SQLite close call fails.
+    pub async fn close(self) -> Result<()> {
+        let (sender, receiver) = oneshot::channel::<std::result::Result<(), SqlError>>();
+
+        if let Err(crossbeam_channel::SendError(_)) = self.sender.send(Message::Close(sender)) {
+            // If the channel is closed on the other side, it means the connection closed successfully
+            // This is a safeguard against calling close on a `Copy` of the connection
+            return Ok(());
+        }
+
+        let result = receiver.await;
+
+        if result.is_err() {
+            // If we get a RecvError at this point, it also means the channel closed in the meantime
+            // we can assume the connection is closed
+            return Ok(());
+        }
+
+        result.unwrap().map_err(|e| AsyncConnError::Close((self, e)))
+    }
+}
+
+impl Debug for AsyncConnection {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result { f.debug_struct("AsyncConnection").finish() }
+}
+
+async fn start<F>(open: F) -> Result<AsyncConnection>
+where
+    F: FnOnce() -> rusqlite::Result<rusqlite::Connection> + Send + 'static,
+{
+    let (sender, receiver) = crossbeam_channel::unbounded::<Message>();
+    let (result_sender, result_receiver) = oneshot::channel();
+
+    thread::spawn(move || {
```

Comment:
Should we use `ctx.spawner()` to spawn this from within our runtime? Currently, I am not sure if this thread will be stopped when using `stop` rpc but I think the receiver will be dropped and the thread will terminate. What do you think?

Reply:
Hmmm worth trying to use our runtime in AsyncConn and check. But for this we have to move `async_sql_conn` files to `mm2_core` crate to access MmCtx and avoid cyclic package dependency error.
Do you  want me to do this check in this PR or in next sprint?

Reply:
> But for this we have to move async_sql_conn files to mm2_core crate to access MmCtx and avoid cyclic package dependency error.

You don't have to that, you can use the `create_subsystem` function
https://github.com/KomodoPlatform/komodo-defi-framework/blob/2c1524dfdcaad68ccabbefc7caa3dfd73e8926d5/mm2src/common/executor/abortable_system/mod.rs#L36
or create a new spawner that uses ctx spawner, e.g.
https://github.com/KomodoPlatform/komodo-defi-framework/blob/2ce476c1ae2be219e5771d8091e7363e264e1579/mm2src/mm2_main/src/lp_native_dex.rs#L529-L530
https://github.com/KomodoPlatform/komodo-defi-framework/blob/a9f606d5ad02e054b5e048eafb37767cb806178c/mm2src/coins/lp_coins.rs#L2587-L2590

> Do you want me to do this check in this PR or in next sprint?

I think it should be checked in this PR since having a detached/dangling thread when using stop is not good , this can lead to resource leakage or even bigger problems.
P.S. I checked our other usage of `thread::spawn` which is done for trezor, it's spawned using mm2 spawner which cause the channel sender to be dropped when stopping mm2 and the event loop to end
https://github.com/KomodoPlatform/komodo-defi-framework/blob/15ff4c59496256c3e85c20ed4c3e4683be6c59dd/mm2src/hw_common/src/transport/libusb.rs#L143
https://github.com/KomodoPlatform/komodo-defi-framework/blob/15ff4c59496256c3e85c20ed4c3e4683be6c59dd/mm2src/hw_common/src/transport/libusb.rs#L151-L152
This is not the case for this the async sql connection thread.

Reply:
As was discussed in dm, added conn close in stop RPC for now [44d93f8](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1989/commits/44d93f891b792705bbf0fdc7ee91510dd905f234)

Reply:
Thanks for the fix! This is an acceptable solution :)